### ISSUE TITLE
ConvertToMySqlDbType for NewDate is not implemented

### ISSUE THREAD

**Issue by dtrickett** [mysql-net/MySqlConnector#1007]
Using MySQL 5.7.12 and running into the following when a UNION query contains a date:

```
  ----> MySqlConnector.MySqlException : Failed to read the result set.
Data:
  Server Error Code: 0
  SqlState: <null>

  ----> System.NotImplementedException : ConvertToMySqlDbType for NewDate is not implemented
```
Seems this is fixed in [MySQL 5.7.22](https://dev.mysql.com/doc/relnotes/mysql/5.7/en/news-5-7-22.html)
"Metadata from result sets for UNION ALL queries could say NEWDATE rather than DATE. (Bug #27422376)"

Is this something that can be addressed within the library, or only by updating MySQL?  We are attempting to migrate from MySql.Data and this is currently the only blocker.

**Comment by bgrainger** [comment]
This will require a bugfix in the library.

Thanks for reporting the detail that it requires an old MySQL 5.7.x version with a `UNION ALL` query to repro.

**Comment by bgrainger** [comment]
Fixed in [1.3.11](https://github.com/mysql-net/MySqlConnector/releases/tag/1.3.11).
